src/include/miopen/conv/context.hpp

```diff
@@ -48,28 +50,23 @@ struct ConvolutionContext : ExecutionContext
     std::string general_compile_options;
 
     ConvolutionContext() = default;
-    ConvolutionContext(conv::Direction dir) : problem(dir) {}
-    ConvolutionContext(const TensorDescriptor& in,
-                       const TensorDescriptor& weights,
-                       const TensorDescriptor& out,
-                       const ConvolutionDescriptor& conv,
-                       conv::Direction dir,
-                       int bias_ = 0)
-        : problem(in, weights, out, conv, dir, bias_)
-    {
-    }
+
+#if MIOPEN_CONV_CONTEXT_USE_FIN_COMPAT_API
     explicit ConvolutionContext(const ProblemDescription& problem_) : problem(problem_) {}
-    ConvolutionContext(const conv::ProblemDescription& problem_, const ExecutionContext& ctx)
-        : ExecutionContext(ctx), problem(problem_)
-    {
-    }
+#endif
+    explicit ConvolutionContext(const ExecutionContext& ctx) : ExecutionContext(ctx) {}
 
-    void SetupFloats();
+#if MIOPEN_CONV_CONTEXT_USE_FIN_COMPAT_API
+    void SetupFloats() { SetupFloats(problem); }
+#endif
+    void SetupFloats(const ProblemDescription& problem);
```

Comment:
I'd suggest moving it to `ProblemDescription` as
```C++
void ProblemDescription::SetupFloats(ExecutionContext& ctx) const
```
To ease removal of ConvolutionContext later on.

Reply:
@DrizztDoUrden I would suggest not making functional changes in this PR so as not to overload it.

Reply:
Okay, not a problem.